src/TimeGutter.js

```diff
@@ -18,8 +18,8 @@ export default class TimeGutter extends Component {
     })
   }
 
-  componentWillReceiveProps(nextProps) {
-    const { min, max, timeslots, step } = nextProps
+  componentDidUpdate() {
+    const { min, max, timeslots, step } = this.props
     this.slotMetrics = this.slotMetrics.update({ min, max, timeslots, step })
```

Comment:
again this too late to update this, because it's consumed in render. it needs to update before render